drivers/SmartThings/zigbee-switch/src/configurations.lua

```diff
@@ -60,8 +66,94 @@ local devices = {
   }
 }
 
+
 local configurations = {}
 
+local active_power_configuration = {
+  cluster = clusters.ElectricalMeasurement.ID,
+  attribute = clusters.ElectricalMeasurement.attributes.ActivePower.ID,
+  minimum_interval = 5,
+  maximum_interval = 3600,
+  data_type = clusters.ElectricalMeasurement.attributes.ActivePower.base_type,
+  reportable_change = 5
+}
+
+local instantaneous_demand_configuration = {
+  cluster = clusters.SimpleMetering.ID,
+  attribute = clusters.SimpleMetering.attributes.InstantaneousDemand.ID,
+  minimum_interval = 5,
+  maximum_interval = 3600,
+  data_type = clusters.SimpleMetering.attributes.InstantaneousDemand.base_type,
+  reportable_change = 5
+}
+
+configurations.check_and_reconfig_devices = function(driver)
+  for device_id, device in pairs(driver.device_cache) do
+    local config_version = device:get_field(CONFIGURATION_VERSION_KEY)
+    if config_version == nil or config_version < driver.current_config_version then
+      if device:supports_capability(capabilities.powerMeter) then
+        if device:supports_server_cluster(clusters.ElectricalMeasurement.ID) then
+          -- Increase minimum reporting interval to 5 seconds
+          device:send(clusters.ElectricalMeasurement.attributes.ActivePower:configure_reporting(device, 5, 600, 5))
+          device:add_configured_attribute(active_power_configuration)
+        end
+        if device:supports_server_cluster(clusters.SimpleMetering.ID) then
+          -- Increase minimum reporting interval to 5 seconds
+          device:send(clusters.SimpleMetering.attributes.InstantaneousDemand:configure_reporting(device, 5, 600, 5))
+          device:add_configured_attribute(instantaneous_demand_configuration)
+        end
+      end
+      device:set_field(CONFIGURATION_ATTEMPTED, true, {persist = true})
+    end
+  end
+  driver._reconfig_timer = nil
```

Comment:
Just want to confirm I'm understanding correctly, the intention is to attempt the config at most once per device each time the driver starts up vs trying periodically in case the device comes back online?

Reply:
Yes, IMO the burden of running a perpetual timer to check all these things is more overhead than it's worth for the edge case of offline devices, and they will be covered the next time the driver restarts.